src/site/layout/css.rs

```diff
@@ -28,27 +35,99 @@ fn concat_minify(css_files: &[String]) -> Result<String> {
     Ok(css)
 }
 
-pub fn get_css_link(
-    dist_dir: &str,
-    path_prefix: &Option<String>,
-    release_tag: &str,
-) -> Result<String> {
-    let filename = fetch_css(dist_dir, release_tag)?;
+pub fn get_css_link(path_prefix: &Option<String>, release_tag: &str) -> Result<String> {
+    let filename = get_css_filename(release_tag);
     Ok(crate::site::link::generate(path_prefix, &filename))
 }
 
-fn fetch_css(dist_dir: &str, release_tag: &str) -> Result<String> {
+/// Places the CSS into the output folder. Depending on if we're running in local development or in
+/// a released binary, the method of obtaining said CSS will differ.
+pub fn place_css(dist_dir: &str, release_tag: &str) -> Result<()> {
+    // Even if you're running a development build, we still respect the custom CSS version preference
+    // by falling back to fetching said version from GitHub.
+    if cfg!(debug_assertions) && release_tag == ORANDA_CSS_TAG {
+        // If we're running in a local environment, we fetch a Tailwind binary and compile the CSS
+        // on the spot. This is useful if we're working on oranda-css locally.
+        build_css(dist_dir)
+    } else {
+        // If we're running in a released binary, we fetch the latest tag (or whichever tag is
+        // specified in the configuration).
+        fetch_css(dist_dir, release_tag)
+    }
+}
+
+pub fn build_css(dist_dir: &str) -> Result<()> {
+    // Fetch our cache dir
+    let project_dir = ProjectDirs::from("dev", "axo", "oranda")
+        .expect("Unable to create cache dir for downloading Tailwind!");
+    let cache_dir = project_dir.cache_dir();
+    // Figure out our target "double" (tailwind has weird naming around this)
+    let double = match (env::consts::OS, env::consts::ARCH) {
+        ("linux", "x86_64") => "linux-x64",
+        ("linux", "aarch64") => "linux-arm64",
+        ("linux", "arm") => "linux-armv7",
+        ("macos", "x86_64") => "macos-x64",
+        ("macos", "aarch64") => "macos-arm64",
+        ("windows", "x86_64") => "windows-x64.exe",
+        ("windows", "aarch64") => "windows-arm64.exe",
+        _ => "linux-x64",
+    };
+    let mut binary_path = Utf8PathBuf::from(cache_dir.display().to_string());
```

Comment:
```suggestion
    let mut binary_path = Utf8PathBuf::from_path(cache_dir)?;
```

Reply:
doesn't work, `Utf8PathBuf` only has `from_path_buf`, which is annoyingly an side-effectful operation

Reply:
(`Utf8Path` has `from_path`, but we need an owned path)